cmd/archive.go

```diff
@@ -51,8 +50,8 @@ An archive is a fully self-contained test run, and can be executed identically e
 			return err
 		}
 		filename := args[0]
-		fs := afero.NewOsFs()
-		src, err := readSource(filename, pwd, fs, os.Stdin)
+		filesystems := createFilesystems()
+		src, err := readSource(filename, pwd, filesystems, os.Stdin)
```

Comment:
Awesome .. I was intending on having this as feature that upgrades the archive to a new version :). This is not actually a problem because after the archive is read it's filesystems are used not the ones that are provided here. Those are used if the file is not an archive but a script

Reply:
Hmm but then it means that we have a potentially multi-megabyte tar archive saved in memory in that cachedFS that won't be garbage-collected. Not a huge issue for `k6 archive`, unless the original tar is huge, but a minor issue for `k6 run` and `k6 cloud`

Reply:
Why would it not be garbage-collected ? We don't save any reference to anything of the fs in ReadSource and newRunner completely ignores it if it's an archive. I don't see why it won't be collected

Reply:
Because the root object of that memory tree would be `filesystems` here (and in the other commands), which persists for the whole duration of the command execution. So, the tar archive will continue to be present there in its `files`, due to the `fsext.CacheOnReadFs` code, I think